catalyst_voices/packages/catalyst_voices_shared/lib/src/platform_widget_builder/platform_widget_builder.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:catalyst_voices_shared/src/platform/catalyst_platform.dart';
+import 'package:flutter/widgets.dart';
+
+class PlatformWidgetBuilder<T> extends StatelessWidget {
+  final Widget Function(BuildContext context, T? data) builder;
+  final T? android;
+  final T? desktop;
+  final T? fuchsia;
+  final T? iOS;
+  final T? linux;
+  final T? macOS;
+  final T? mobile;
+  final T? mobileWeb;
+  final T? web;
+  final T? webDesktop;
+  final T? windows;
+  final T? other;
+
+  const PlatformWidgetBuilder({
+    super.key,
+    required this.builder,
+    this.android,
+    this.desktop,
+    this.fuchsia,
+    this.iOS,
+    this.linux,
+    this.macOS,
+    this.mobile,
+    this.mobileWeb,
+    this.web,
+    this.webDesktop,
+    this.windows,
+    this.other,
+  });
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
What if we introduce a `Map` in `CatalystPlatform` like this:

```dart
  static Map<String, bool> get identifiers => {
        'android': isAndroid,
        'desktop': isDesktop,
        'fuchsia': isFuchsia,
        'iOS': isIOS,
        'linux': isLinux,
        'macOS': isMacOS,
        'mobile': isMobile,
        'mobileWeb': isMobileWeb,
        'web': isWeb,
        'webDesktop': isWebDesktop,
        'windows': isWindows,
      };
}
```

Then, we can eliminate all `if else` statements and compare two dictionaries. 
The class would look something like this: 

```dart
class PlatformAwareBuilder<T> extends StatelessWidget {
  final Widget Function(BuildContext context, T? data) builder;
  final Map<String, T?> _platformData;

  PlatformAwareBuilder({
    super.key,
    required this.builder,
    T? android,
    T? desktop,
    T? fuchsia,
    T? iOS,
    T? linux,
    T? macOS,
    T? mobile,
    T? mobileWeb,
    T? isWeb,
    T? webDesktop,
    T? windows,
    T? other,
  }) : _platformData = {
          'android': android,
          'desktop': desktop,
          'fuchsia': fuchsia,
          'iOS': iOS,
          'linux': linux,
          'macOS': macOS,
          'mobile': mobile,
          'mobileWeb': mobileWeb,
          'web': isWeb,
          'webDesktop': webDesktop,
          'windows': windows,
          'other': other,
        };

  @override
  Widget build(BuildContext context) {
    return builder(context, _getPlatformData());
  }

  T _getPlatformData() {
    final currentPlatformKey = CatalystPlatform.identifiers.entries
        .firstWhere(
          (entry) => entry.value,
          orElse: () => const MapEntry('Other', true),
        )
        .key;
    return _platformData[currentPlatformKey]!;
  }
}
```

As an additional improvement, we should consider extracting `String` keys for `Map`s `identifiers` and `_platformData` to separate `Class` or `enum` and re-use it.

@coire1  What do you think?

Reply:
OK, I like it. I will refactor it.